[jednghk] iP

jednhk opened the pull request:
### Duke

> Some people look good from far, but are far from good. ([Source](me.com))

Have you ever wanted to:

1. Not use **much nicer UIs** for your event planning :thinking:
2. Make your life harder?

Then this is the app for you!

As a software engineering project done by an inexperienced programmer, Duke can:

- Record your todos, deadlines, events, but only when you follow a very specific format
- That's about it!

- [ ] Here's a checkbox for you to click for fun

If you're reading `this far` for some reason, here's a code block for you to analyse:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

jednhk committed:
Add an echo feature for Duke skeleton

jednhk committed:
Implement addTask and displayTasks feature

jednhk committed:
Create Main class, Task class and TaskList

jednhk committed:
Add Todo, Event, Deadline classes

jednhk committed:
Add InvalidCommandException, ZeroLengthDescriptionException

bokuanT commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,42 @@
+        taskToMark.unmark();
+    }
+
+    public String getTaskList() {
```
Just picking a nit but maybe you could consider renaming this method to `getTaskListString`. Might be less confusing in the future if you ever need a method to return the `List<Task>` `taskList`.

bokuanT commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,87 @@
+        while (true) {
+            String userInput = sc.nextLine().toLowerCase();
+            String[] command = userInput.split(" ", 10);
+            switch (command[0]) {
```
It's good that you adhered to the CS2103 java coding standards for switch statements.

bokuanT commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,42 @@
+import java.util.List;
+
+public class TaskList {
+    private List<Task> taskList = new LinkedList<>();
```
Was there a particular reason why you chose to use a `LinkedList<E>` over an `ArrayList<E>`?

bokuanT commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,87 @@
+                + "____________________________________________________________");
+    }
+
+    private void greet() {
```
Would you consider encapsulating all of Duke's interactions with the user into a separate class? The class could be the `Ui` class, containing the methods `greet()`, `printExitMessage()`, etc.

bokuanT reviewed (commented):
LGTM. 👍 
Coding standards are adhered to well and I like how you are making use of OOP by extracting the task list into the `TaskList` class.

jednhk committed:
Add Automated text UI testing

jednhk committed:
Add Save and Load feature for tasks

jednhk committed:
Add date parser to format deadline and event dates

jednhk committed:
Merge branch 'branch-Level-7'

jednhk committed:
Extract Parser, Ui, Storage classes

jednhk committed:
Merge branch 'branch-Level-8'

jednhk committed:
Extract methods previously in Duke to respective classes (Ui, Storage or Parser)

jednhk committed:
Move all files into duke package

jednhk committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

jednhk committed:
Add 2 JUnit tests for todo command

jednhk committed:
Add JavaDoc comments

jednhk committed:
Add find features to get tasks containing a keyword

CedricPei commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,92 @@
+            }
+        }
+    }
+}
```
For the project structure, I think it will be good to move Tasks (Deadline Todo...) to a package and move others to another.

CedricPei replied:
Also the main class should look more concise, the code inside the while loop could be extracted to Parser or another class.

CedricPei commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,96 @@
+import java.io.*;
+/** Handles the storage and retrieval of tasks. */
+public class Storage {
+    TaskList taskList;
```
better to add public/private

CedricPei commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,54 @@
+            Event event = (Event) this;
+            return String.format("%c|%b|%s|%s|%s", taskType, isDone, taskName, event.getFrom(), event.getTo());
+        }
+    }
```
For this part, to use a switch-case block is concise

CedricPei commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,55 @@
+    public void unmarkTask(int taskIndex) {
+        Task taskToMark = this.getTask(taskIndex);
+        taskToMark.unmark();
+    }
```
more comments expected for the methods above

jednhk committed:
Add relevant checkstyle files and ensure coding standard is adhered to

jednhk committed:
Set up JavaFX

jednhk committed:
Implement GUI for Duke

jednhk committed:
Add Assertion statements

jednhk committed:
Add more assertion statements

jednhk committed:
Merge pull request #2 from jednghk/branch-A-Assertions

Add more assertion statements

jednhk committed:
Abstract getResponse() method body

The getResponse() method body does not adhere to SLAP and has an arrowhead style code. This makes the code messier and harder to debug.

Refactoring the method body will improve code quality.

Let's create a Command class which handles the execution of the user input, and split the code into their respective methods to mitigate the arrowhead style code.

jednhk committed:
Merge pull request #3 from jednghk/branch-A-CodeQuality

Abstract getResponse() method body

jednhk committed:
Update README.md

jednhk committed:
Update README.md

jednhk committed:
Update user guide

jednhk committed:
Merge branch 'master' of https://github.com/jednghk/ip

jednhk committed:
Update user guide to include all features

jednhk committed:
Change file paths to be relative
